src/overload_ifelse.jl

```diff
@@ -0,0 +1,41 @@
+const SimpleTracer = Union{ConnectivityTracer,GradientTracer,HessianTracer}
+
+function Base.ifelse(::T, x, y) where {T<:SimpleTracer}
+    size(x) != size(y) &&
+        throw(DimensionMismatch("Outputs sizes of `ifelse` arguments don't match in size."))
+    return output_union(x, y)
+end
+
+## output union on scalar outputs
+output_union(tx::C, ty::C) where {C<:ConnectivityTracer} = C(inputs(tx) ∪ inputs(ty))
+output_union(tx::G, ty::G) where {G<:GradientTracer} = G(gradient(tx) ∪ gradient(ty))
+function output_union(tx::H, ty::H) where {H<:HessianTracer}
+    return H(gradient(tx) ∪ gradient(ty), hessian(tx) ∪ hessian(ty))
+end
+
+output_union(tx::T, y) where {T<:SimpleTracer} = tx
+output_union(x, ty::T) where {T<:SimpleTracer} = ty
```

Comment:
be careful to have something more specific for `(Tracer, Dual)` and `(Dual, Tracer)` otherwise this will take precedence and return incorrect results

Reply:
This doesn't work with `Dual`.

Reply:
I meant that in the first function, `y` could be a `Dual`, and in the second `x` could be a `Dual`

Reply:
Two different tracer types never appear together.
We either create `ConnectivityTracer`, `GradientTracer`, `HessianTracer` or `Dual`. They never appear together, unless we made a huge mistake somewhere.